ingest/ledger_transaction.go

```
@@ -181,3 +188,413 @@ func (t *LedgerTransaction) operationChanges(ops []xdr.OperationMeta, index uint
 func (t *LedgerTransaction) GetDiagnosticEvents() ([]xdr.DiagnosticEvent, error) {
 	return t.UnsafeMeta.GetDiagnosticEvents()
 }
+
+func (t *LedgerTransaction) ID() int64 {
+	return toid.New(int32(t.Ledger.LedgerSequence()), int32(t.Index), 0).ToInt64()
+}
+
+func (t *LedgerTransaction) Account() (string, error) {
+	sourceAccount := t.Envelope.SourceAccount()
+	providedID := sourceAccount.ToAccountId()
+	pointerToID := &providedID
+
+	return pointerToID.GetAddress()
+}
+
+func (t *LedgerTransaction) AccountSequence() int64 {
+	return t.Envelope.SeqNum()
+}
+
+func (t *LedgerTransaction) MaxFee() uint32 {
+	return t.Envelope.Fee()
+}
+
+func (t *LedgerTransaction) FeeCharged() (int64, bool) {
+	// Any Soroban Fee Bump transactions before P21 will need the below logic to calculate the correct feeCharged
+	// Protocol 20 contained a bug where the feeCharged was incorrectly calculated but was fixed for
+	// Protocol 21 with https://github.com/stellar/stellar-core/issues/4188
+	var ok bool
+	_, ok = t.GetSorobanData()
+	if ok {
+		if uint32(t.Ledger.LedgerHeaderHistoryEntry().Header.LedgerVersion) < 21 && t.Envelope.Type == xdr.EnvelopeTypeEnvelopeTypeTxFeeBump {
+			var resourceFeeRefund int64
+			var inclusionFeeCharged int64
+
+			resourceFeeRefund, ok = t.SorobanResourceFeeRefund()
+			if !ok {
+				return 0, false
+			}
+
+			inclusionFeeCharged, ok = t.SorobanInclusionFeeCharged()
+			if !ok {
+				return 0, false
+			}
+
+			return int64(t.Result.Result.FeeCharged) - resourceFeeRefund + inclusionFeeCharged, true
+		}
+	}
+
+	return int64(t.Result.Result.FeeCharged), true
+
+}
+
+func (t *LedgerTransaction) OperationCount() uint32 {
+	return t.Envelope.OperationsCount()
+}
+
+func (t *LedgerTransaction) Memo() string {
+	memoObject := t.Envelope.Memo()
+	memoContents := ""
+	switch xdr.MemoType(memoObject.Type) {
```

I think you should handle all cases including MEMO_NONE and then you can add a default case which panics. That way if you forget to support a new memo case in the future the library can fail instead of mistakenly returning the empty string without errors

---

Updated in 872137ec1600d55d9f84e56adf2a86d84381a3bf